docs/notebooks/model_config.pct.py

```diff
@@ -110,19 +113,79 @@ def build_model(data):
 }
 
 # %% [markdown]
-# Next you simply pass the configuration dictionary to the `optimize` function and `BayesianOptimizer` will sort out which model and optimizer interface needs to be used to build a valid model.
+# Next you simply pass the configuration dictionary to the `optimize` function and `BayesianOptimizer` will sort out which model and optimizer wrapper needs to be used to build a valid model.
 
 # %%
 bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
 result = bo.optimize(2, initial_data, model_config)
 
 
+# %% [markdown]
+# ## Using configuration dictionaries for setting up experiments
+#
+# Another use case is in setting up experiments, where it becomes easier to benchmark Bayesian optimization algorithms. The advantage is that we can easily change the models and set up any argument for them from one condition to another - only object with the experiment specification needs to change (`experiment_conditions` below), while the rest of the code for executing experiments can stay the same. 
+
+# %%
+from copy import deepcopy
+
+from trieste.models.gpflux import DeepGaussianProcess, build_vanilla_deep_gp
+
+
+def build_gp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.Matern52(variance=variance, lengthscales=[0.2, 0.2])
+    model = GPR(data.astuple(), kernel, noise_variance=1e-5)
+    return model
+
+
+def build_dgp_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    dgp = build_vanilla_deep_gp(data.query_points, num_layers=2, num_inducing=100)
+    dgp.f_layers[-1].kernel.kernel.variance.assign(variance)
+    dgp.f_layers[-1].mean_function = gpflow.mean_functions.Constant()
+    dgp.likelihood_layer.likelihood.variance.assign(1e-5)
+    return dgp
+
+
+def run_experiment(model_config):
+    bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
+    result = bo.optimize(2, initial_data, model_config)
+    return result
+
+
+# configuration shared by all experiments, this is modified by each experiment condition
+basic_config = {
+    "model": gpflow_model,
+    "model_args": {
+        "num_kernel_samples": 100,
+    },
+    "optimizer": gpflow.optimizers.Scipy(),
+    "optimizer_args": {
+        "minimize_args": {"options": dict(maxiter=100)},
+    },
+}
+
+# here we specify our experiments
+experiment_conditions = [
+    {"model_args": {"num_kernel_samples": 500}},
+    {"model_args": {"num_kernel_samples": 1000}},
+    {"model": build_dgp_model(initial_data)},
+]
```

Comment:
To make sure this doesn't affect CI you may want to leave only one, or none at all, for quick test purposes.

Reply:
Agree. Feel free to ask me if you're unsure how to set up the quicktest overrides.

Reply:
gave it a try, @uri-granta how can I check whether quickrun is working properly?

Reply:
Maybe you could try running quicktest.py locally and see what notebook it creates?

Reply:
its not working, probably regular expression in the quickrun.py is not that flexible, I suggest we leave it as is, its two BO runs now with 2 iterations only, so should be quick enough